internal/providers/terraform/terragrunt_provider.go

```diff
@@ -69,7 +69,16 @@ func (p *TerragruntProvider) DisplayType() string {
 }
 
 func (p *TerragruntProvider) AddMetadata(metadata *schema.ProjectMetadata) {
-	// no op
+	basePath := p.ctx.ProjectConfig.Path
+	if p.ctx.RunContext.Config.ConfigFilePath != "" {
+		basePath = filepath.Dir(p.ctx.RunContext.Config.ConfigFilePath)
+	}
+
+	modulePath, err := filepath.Rel(basePath, metadata.Path)
+	if err == nil && modulePath != "" && modulePath != "." {
+		log.Debugf("Calculated relative terraformModulePath for %s from %s", basePath, metadata.Path)
+		metadata.TerraformModulePath = modulePath
+	}
```

Comment:
Does it make sense to add `metadata.TerraformWorkspace = p.ctx.ProjectConfig.TerraformWorkspace` like in other places for uniformity?